### PR TITLE
Add constellation

### PR DESCRIPTION
**Pull request by brendanzab**
@kvark mentioned in his presentation to the Toronto Rust meetup that he didn't know how [constellation](https://github.com/TomGillen/constellation/) fared in comparison to [specs](https://github.com/slide-rs/specs). So I endeavored to find out!

cc. @TomGillen

### CODE REVIEW

**Comment by brendanzab** [comment]
The numbers are a little suspicious, at least for me:

- pos_vel update
  - constellation: 9,210 ns/iter (+/- 1,777)
  - froggy: 15,081 ns/iter (+/- 3,146)
  - specs: 45,860 ns/iter (+/- 15,550)
- parallel update
  - constellation: 164,793 ns/iter (+/- 29,864)
  - froggy: 66,002 ns/iter (+/- 21,315)
  - specs: 57,347 ns/iter (+/- 11,922)

That's a biiig slow down on the parallel update... 😕

**Comment by TomGillen** [comment]
There are a few things I have to note about this.

1. You shouldn't be constructing a new `SystemCommandBuffer` every iteration. The purpose of that buffer is to precompute system scheduling so that it does not need to be performed every frame.
2. If you want to force sequential scheduling for the sequential test, then use `world.run_sequential(&mut update)` rather than `world.run(&mut update)`.
3. The parallel test will actually run sequentially, as the parallelism is performed at the system level and you are only running one system. However, it should be possible to nest parallelism inside a system. I'll need to investigate if it is possible to have the iter_* functions do this automatically.

File: benches/parallel_constellation.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+#[bench]
+fn bench_update(b: &mut Bencher) {
+    let mut world = build();
+
+    b.iter(|| {
+        let mut update = SystemCommandBuffer::default();
+        update.queue_systems(|scope| {
+            scope.run_r1w2(|ctx, rs: &Rs, w1s: &mut W1s, w2s: &mut W1s| {
```
Comment (TomGillen) [inline benches/parallel_constellation.rs:52]: The type of w2s should be `&mut W2s`?

**Comment by kvark** [comment]
Nice PR! Constellation is pretty impressive 👍

File: benches/parallel_constellation.rs (hunk 1 above)
Comment (lschmierer) [reply benches/parallel_constellation.rs:52]: I am going to fix this

Comment (lschmierer) [reply benches/parallel_constellation.rs:52]: @TomGillen I tried fixing it on #19 but it seems to run endless now and also the "building" takes very long

**Pull request merged**